core/platform/models.py

```diff
@@ -60,11 +60,167 @@ def import_models(cls):
+            external associations have been deleted.
+        """
+        return (
+            self._impl.verify_external_auth_associations_are_deleted(user_id))
+
+    def get_auth_id_from_user_id(self, user_id):
+        """Returns the auth ID associated with the given user ID.
+
+        Args:
+            user_id: str. The user ID.
+
+        Returns:
+            str|None. The auth ID associated with the given user ID, or None if
+            no association exists.
+        """
+        return self._impl.get_auth_id_from_user_id(user_id)
+
+    def get_user_id_from_auth_id(self, auth_id):
+        """Returns the user ID associated with the given auth ID.
+
+        Args:
+            auth_id: str. The auth ID.
+
+        Returns:
+            str|None. The user ID associated with the given auth ID, or None if
+            no association exists.
+        """
+        return self._impl.get_user_id_from_auth_id(auth_id)
+
+    def get_multi_user_ids_from_auth_ids(self, auth_ids):
+        """Returns the user IDs associated with the given auth IDs.
+
+        Args:
+            auth_ids: list(str). The auth IDs.
+
+        Returns:
+            list(str|None). The user IDs associated with each of the given auth
+            IDs, or None for associations which don't exist.
+        """
+        return self._impl.get_multi_user_ids_from_auth_ids(auth_ids)
+
+    def get_multi_auth_ids_from_user_ids(self, user_ids):
+        """Returns the auth IDs associated with the given user IDs.
+
+        Args:
+            user_ids: list(str). The user IDs.
+
+        Returns:
+            list(str|None). The auth IDs associated with each of the given user
+            IDs, or None for associations which don't exist.
+        """
+        return self._impl.get_multi_auth_ids_from_user_ids(user_ids)
+
+    def associate_auth_id_with_user_id(self, auth_id_user_id_pair):
+        """Commits the association between auth ID and user ID.
+
+        Args:
+            auth_id_user_id_pair: auth_domain.AuthIdUserIdPair. The association
+                to commit.
+
+        Raises:
+            Exception. The IDs are already associated with a value.
+        """
+        self._impl.associate_auth_id_with_user_id(auth_id_user_id_pair)
+
+    def associate_multi_auth_ids_with_user_ids(self, auth_id_user_id_pairs):
+        """Commits the associations between auth IDs and user IDs.
+
+        Args:
+            auth_id_user_id_pairs: list(auth_domain.AuthIdUserIdPair). The
+                associations to commit.
+
+        Raises:
+            Exception. One or more auth associations already exist.
+        """
+        self._impl.associate_multi_auth_ids_with_user_ids(auth_id_user_id_pairs)
+
+
 class _Gae(Platform):
     """Provides platform-specific imports related to
     GAE (Google App Engine).
     """
 
+    def __init__(self):
+        # NOTE: We can't initialize this value yet, because if we do it would
```

Comment:
I think it would be better for the platform layer not to import from this module. It seems like the only thing that gae_auth_services and firebase_auth_services imports is transaction_services ... but it doesn't use it at all. So can we get rid of it entirely and thus remove this special casing?

Reply:
The issue is in test_utils which creates models and uses services, so that wouldn't help here.